app/javascript/packs/MainApp/components/RefereeProfile.jsx

```diff
@@ -12,43 +14,59 @@ class RefereeProfile extends Component {
   state = {
     httpStatus: 0,
     httpStatusText: '',
-    firstName: '',
-    lastName: '',
-    bio: '',
-    email: '',
-    showPronouns: false,
-    pronouns: '',
-    nationalGoverningBodies: [],
-    certifications: []
+    referee: {
+      firstName: '',
+      lastName: '',
+      bio: '',
+      email: '',
+      showPronouns: false,
+      pronouns: '',
+      nationalGoverningBodies: [],
+      certifications: [],
+      isEditable: false
+    },
+    edit: false
   };
 
   componentDidMount() {
-    const { match: { params } } = this.props;
-
-    axios.get(`/api/v1/referees/${params.id}`)
+    axios
+      .get(this.getCurrentRefereeApiRoute())
       .then(this.setComponentStateFromBackendData.bind(this))
       .catch(this.setErrorStateFromBackendData.bind(this))
   }
 
+  getCurrentRefereeApiRoute() {
+    const { match: { params } } = this.props;
+
```

Comment:
excellent abstraction, let's take this a step further by using a getter function in the class:

```
get currentRefereeApiRoute () {
  const { match: { params } } = this.props

  return `/api/v1/referees/${params.id}`
}
```
This is another "memoization" strategy that reduces js calculations